lib/poller.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+class Poller {
+  constructor ({
+    pollingRateSeconds,
+    serializeSegments,
+    splitioApi
+  }) {
+    this.cache = {}
+    this.pollingRateSeconds = pollingRateSeconds || 300
+    this.serializeSegments = serializeSegments || false
+    this.splitioApi = splitioApi
+  }
+
+  async pollForChanges () {
+    try {
+      const splitChanges = await this.splitioApi.getSplitChanges()
+      Object.assign(this.cache, splitChanges)
```

Comment:
should you handle removal here?  otherwise there's a risk the cache continues to grow.

Reply:
removed splits won't be returned with since=-1. and this PR filters them out on subsequent requests: https://github.com/godaddy/split-node-serializer/pull/10/files